### PR TITLE
Force with auto grad

### PR DESCRIPTION
**Pull request by mcsloy**
Replaced the current explicit and manual method for evaluating atomic forces with a PyTorch auto-grad implementation. This pull request also removes the original repulsive spline class.

### CODE REVIEW

**Comment by mcsloy** [comment]
Note that I have not placed any check or warning associated with using the "last-step" DFTB-SCC gradient method when computing the forces. This is because I am finding that the last-step method tends to produce acceptable forces, not as accurate as those produced using the "direct" or "implicit" methods but good enough (typically one order of magnitude larger). A script demonstrating this can be found [here](https://gist.github.com/mcsloy/2137d102bf4d57cd7d154e92592951dc).

File: tbmalt/physics/dftb/__init__.py [hunk 1]
```diff
@@ -672,6 +492,33 @@ class Dftb2(Calculator):
         filling_scheme: The scheme used for finite temperature broadening.
             There are two broadening methods, Fermi-Dirac broadening and
             Gaussian broadening, supported in TBMaLT. [DEFAULT: fermi]
+        grad_mode: controls gradient mode is used when running the SCC
```
Comment (WbSun723) [inline tbmalt/physics/dftb/__init__.py:495]: controls gradient mode ~~is~~ used when running the SCC

File: tbmalt/physics/dftb/__init__.py [hunk 2]
```diff
@@ -672,6 +492,33 @@ class Dftb2(Calculator):
...
+            cycles. Available options are [DEFAULT:"last_step"]:
+
+                - "direct": run the SCC cycle with without special treatment.
```
Comment (WbSun723) [inline tbmalt/physics/dftb/__init__.py:498]: run the SCC cycle ~~with~~ without special treatment.

File: tests/unittests/test_implicit.py [hunk 3]
```diff
@@ -61,20 +65,21 @@ def dftb_calculator(device, skf_file: str):
...
 
     # set up the calculator
     return Dftb2(hamiltonian_feed, overlap_feed, occupation_feed,
                  hubbard_feed, r_feed=repulsive_feed, filling_scheme=None)
 
 
 def implicit_gradient_helper(mol: Geometry, orbs: OrbitalInfo, dftb_calculator, device):
-
+    dftb_calculator.grad_mode = "direct"
     energy_direct = dftb_calculator(mol, orbs, grad_mode='direct')
```
Comment (WbSun723) [inline tests/unittests/test_implicit.py:77]: energy_direct = dftb_calculator(mol, orbs, ~~grad_mode='direct'~~)
Hey Adam, as `grad_mode` now is an argument of the calculator, we may not need to specify it during the forward calling.

File: tests/unittests/test_implicit.py [hunk 4]
```diff
@@ -61,20 +65,21 @@ def dftb_calculator(device, skf_file: str):
...
     forces_direct = -torch.autograd.grad(
         energy_direct, mol.positions,
         grad_outputs=torch.ones_like(energy_direct))[0]
 
+    dftb_calculator.grad_mode = "implicit"
     energy_imp = dftb_calculator(mol, orbs, grad_mode='implicit')
```
Comment (WbSun723) [inline tests/unittests/test_implicit.py:83]: energy_imp = dftb_calculator(mol, orbs, ~~grad_mode='implicit'~~)

Comment (WbSun723) [reply tests/unittests/test_implicit.py:83]: Hey Adam, since we have changed `grad_mode` as an argument of the calculator, we also need to change corresponding definitions of calculators in our examples.
For example: 
https://github.com/mcsloy/tbmalt_new/blob/29b657958c6e721e9af9c683339d2c529f346821/examples/example_01/example_01.py#L98
https://github.com/mcsloy/tbmalt_new/blob/29b657958c6e721e9af9c683339d2c529f346821/examples/example_01/example_01.py#L176
I can make a new pull request later to fix it :)

Comment (mcsloy) [reply tests/unittests/test_implicit.py:83]: Ah yes, you are right, there are a few of those that I have missed in various places. I will remove the `**kwargs` option from the `forward` method so that unknown arguments raise an error, and I will locate and remove all instance where `grad_mode` is specified in the calculator run call.
